core/src/test/scala/filodb.core/memstore/PartKeyLuceneIndexSpec.scala

```diff
@@ -786,6 +786,28 @@ class PartKeyLuceneIndexSpec extends AnyFunSpec with Matchers with BeforeAndAfte
     keyIndex.singlePartKeyFromFilters(Seq(filter2), 4, 10) shouldBe None
   }
 
+  it("should get a single match for part keys by a regex filter") {
```

Comment:
Could you add a couple tests that just call `removeRegexAnchors` to confirm visually that things like:
```
^$foo.*$\$$
foo.*\\\$
foo.*$\\\\$
```
etc. are updated as expected?

Reply:
Added a test for them